### PR TITLE
Add Lint CI

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

**Comment by fbeutin-ledger** [comment]
Lint errors are fixed by this PR:
https://github.com/LedgerHQ/app-solana/pull/34
It will need to be merged first in order to have green CI

The PRs have been separated in order to not drown the useful content

**Comment by fbeutin-ledger** [comment]
Build debug CI have been matrixed so the CI run requirement will never complete
The branch protection will need to be updated once this PR is merged

File: .github/workflows/lint-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  job_lint:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clone
+        uses: actions/checkout@v2
+
+      - name: Lint
+        uses: DoozyX/clang-format-lint-action@v0.11
```
Comment (apaillier-ledger) [inline .github/workflows/lint-workflow.yml:24]: This should be version 0.13

File: .github/workflows/lint-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          source: './src'
+          extensions: 'h,c'
+          clangFormatVersion: 11
```
Comment (apaillier-ledger) [inline .github/workflows/lint-workflow.yml:28]: This should use clang version 12.0.0

**Review by apaillier-ledger — DISMISSED** [review]

**Pull request merged**